logstash-core/benchmarks/src/main/java/org/logstash/benchmark/QueueRWBenchmark.java

```diff
@@ -0,0 +1,120 @@
+import org.logstash.ackedqueue.io.MmapPageIO;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@Warmup(iterations = 3, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+@Fork(1)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
+public class QueueRWBenchmark {
+
+    private static final int EVENTS_PER_INVOCATION = 500_000;
+    
+    private static final int BATCH_SIZE = 100;
+
+    private static final Event EVENT = new Event();
+
+    private Queue queue;
+
+    private String path;
+    
+    private ExecutorService exec;
+
+    @Setup
+    public void setUp() throws IOException, CloneNotSupportedException {
+        final Settings settings = settings();
+        EVENT.setField("Foo", "Bar");
+        EVENT.setField("Foo1", "Bar1");
+        EVENT.setField("Foo2", "Bar2");
+        EVENT.setField("Foo3", "Bar3");
+        EVENT.setField("Foo4", "Bar4");
+        path = settings.getDirPath();
+        queue = new Queue(settings);
+        queue.open();
+        exec = Executors.newSingleThreadExecutor();
+    }
+
+    @TearDown
+    public void tearDown() throws IOException {
+        queue.close();
+        FileUtils.deleteDirectory(new File(path));
+        exec.shutdownNow();
+    }
+
+    @Benchmark
+    @OperationsPerInvocation(EVENTS_PER_INVOCATION)
+    public final void readFromPersistedQueue(final Blackhole blackhole) throws Exception {
+        final Future<?> future = exec.submit(() -> {
+            for (int i = 0; i < EVENTS_PER_INVOCATION; ++i) {
+                try {
+                    final Event evnt = EVENT.clone();
+                    evnt.setTimestamp(Timestamp.now());
+                    this.queue.write(evnt);
+                } catch (final IOException | CloneNotSupportedException ex) {
+                    throw new IllegalStateException(ex);
+                }
+            }
+        });
+        for (int i = 0; i < EVENTS_PER_INVOCATION / BATCH_SIZE; ++i) {
+            final Batch batch = queue.readBatch(BATCH_SIZE);
+            for (final Queueable elem : batch.getElements()) {
+                blackhole.consume(elem);
+            }
+        }
```

Comment:
Fixed :)

performance hit is in the margin of error which is a good thing I guess:

now

```
Benchmark                                  Mode  Cnt   Score    Error   Units
QueueRWBenchmark.readFromPersistedQueue   thrpt   10  45.492 ± 10.648  ops/ms
QueueWriteBenchmark.pushToPersistedQueue  thrpt   10  54.942 ± 13.988  ops/ms
```